mmv1/templates/terraform/encoders/backend_service.go.erb

```diff
@@ -47,4 +47,43 @@ for _, backendRaw := range backends {
 	}
 }
 
+// This custom encoding helps prevent sending 0 for clientTtl, defaultTtl and
```

Comment:
Similar to the referenced PR, could we add or modify a test case so that it failed or had a permadiff prior to the change and doesn't now?

Reply:
I've added a test now. What made it hard for me to write and run the test was that when I built the provider using the command:
```
make provider VERSION=ga OUTPUT_PATH="$GOPATH/src/github.com/hashicorp/terraform-provider-google" PRODUCT=compute
```
And then ran the test using:
```
make testacc TEST=./google/services/compute TESTARGS='-run=TestAccComputeBackendService_withCdnPolicy'
 ```
 I got the "error":
 ```
 testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-google/google/services/compute  0.602s [no tests to run]
```
It confused me whether I was doing the right thing, and as I could run the existing tests, I didn't know if and how to change the tests and verify them properly.

--- 

After some digging today, I found out that the tests seem to not be included/run in the ga version of the module, and thus, I could run them.
<img width="503" alt="image" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/1aa98657-1a1e-4772-918d-4a0be51893e6">

--- 

@rileykarson is there a reason why the tests do not run in the ga version (and only in the beta version?)?
This was a bit confusing in the development experience for me.
Anyhow I've added the tests now, they did not pass at first but after the change to the encoder they do.

Reply:
That test is wrapped in a [version guard](https://googlecloudplatform.github.io/magic-modules/develop/promote-to-ga/#promote-from-beta-to-ga), which generates the test in just the beta provider. They're used to wrap tests using fields or resources only present in the beta provider. It's likely that the guards have just drifted- that someone promoted the beta fields that were in use and didn't remember to update the test. If the test passes in GA when they're removed, feel free to remove them! You should just have to [remove the guards](https://googlecloudplatform.github.io/magic-modules/develop/promote-to-ga/#promote-tests)

Reply:
I've ran the tests in the ga version and it worked, removing the version guard.